applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1342,6 +1421,42 @@ static bool handle_hid_report_sent_event(
 	return false;
 }
 
+static void update_keyboard_leds(struct subscriber *sub, uint8_t kbd_leds)
+{
+	if (sub->kbd_leds != kbd_leds) {
+		sub->kbd_leds = kbd_leds;
+
+		if (sub == get_keyboard_subscriber()) {
+			broadcast_keyboard_leds(sub);
+		}
+	}
+}
+
+static bool handle_hid_report_event(const struct hid_report_event *event)
+{
+	if (event->dyndata.data[0] != REPORT_ID_KEYBOARD_LEDS) {
```

Comment:
I wonder if we should actually handle all of the HID output reports in HID state. The HID keyboard LEDs report is a "special" case since it's tightly coupled with HID keyboard keys report. We currently do not support any other HID output report (except the one used for config channel).

Reply:
If we would like to use more generic approach then maybe we should simply move this part to a separate module to avoid complicating HID state? The only thing we would need to duplicate is tracking subscriptions.

Reply:
> I wonder if we should actually handle all of the HID output reports in HID state

I don't know it either but writing this function in a generic way costs us nothing. The only difference is name and a fact that `REPORT_ID_KEYBOARD_LEDS` is hidden behind an argument.

> If we would like to use more generic approach then maybe we should simply move this part to a separate module

Making this function generic will not add any complexity to what is already here. We could think of splitting hid_state but not during this PR and not in the near future.

Reply:
Discussed offline. I will add more generic implementation where possible.